apps/meteor/app/integrations/server/api/api.js

```diff
@@ -237,6 +240,51 @@ function integrationInfoRest() {
 }
 
 class WebHookAPI extends APIClass {
+	async authenticatedRoute(request) {
+		const payloadKeys = Object.keys(request.body);
+		const payloadIsWrapped = request.body && request.body.payload && payloadKeys.length === 1;
+		if (payloadIsWrapped && request.headers['content-type'] === 'application/x-www-form-urlencoded') {
+			try {
+				request.body = JSON.parse(request.body.payload);
+			} catch ({ message }) {
+				return {
+					error: {
+						statusCode: 400,
+						body: {
+							success: false,
+							error: message,
+						},
+					},
+				};
+			}
+		}
+
+		request.integration = await Integrations.findOne({
+			_id: request.params.integrationId,
+			token: decodeURIComponent(request.params.token),
+		});
```

Comment:
Should we have this one on model? Can be done later, a TODO may be enough.
But if we can do it on here :)

Reply:
Not sure if I followed, do you mean create a 'getByIdAndToken' ? I dont think is  related to the objetive of the pull request, this is 100% exactly how the previous code was doing

Reply:
Yup, that's the idea. Agree on the oos, that's why I suggested the TODO. I'm tracking and fixing some of them.